app/src/main/java/org/simple/clinic/instantsearch/InstantSearchUpdate.kt

```diff
@@ -13,8 +14,11 @@ import org.simple.clinic.mobius.next
 import org.simple.clinic.patient.OngoingNewPatientEntry
 import org.simple.clinic.patient.PatientSearchCriteria
 import org.simple.clinic.patient.businessid.Identifier
+import javax.inject.Inject
 
-class InstantSearchUpdate : Update<InstantSearchModel, InstantSearchEvent, InstantSearchEffect> {
+class InstantSearchUpdate @Inject constructor(
+    private val features: Features
```

Comment:
We should not be passing the `Features` type to any `Update` function because it makes the `Update` function impure. What we should be doing instead is only taking the feature flag we need (in this case, the `InstantSearchByPatientIdentifier` feature and passing it as a `Boolean` property to the `Update` function.

Then we use the `Features` type in the screen at the time of instantiating the `Update` class to get the actual flag value.